### ISSUE TITLE
False positive unused_variables on never-type

### ISSUE THREAD

**Issue by A4-Tacks** [rust-lang/rust-analyzer#22975]
**rust-analyzer version**: rust-analyzer 0.4.2991-standalone

**rustc version**: rustc 1.97.0-nightly (36ba2c771 2026-04-23)

**editor or extension**: Vim

**code snippet to reproduce**:
```rust
fn main() {
    let foo = loop {};
    _ = foo; // used
}
```

Cursor on `foo` (line 2)

Expect no this quickfix, but actual output `Rename unused foo to _foo`

After apply:

```rust
fn main() {
    let _foo = loop {};
    _ = foo; // used
}
```

Should either 'not applicable' or 'rename all usage'
